lib/src/package_analysis/lower_bound_constraint_analysis.dart

```diff
@@ -0,0 +1,630 @@
+            constraint: context.dependencies[packageName]!.version,
+            currentVersion: context.findInstalledVersion(packageName),
+            lowestVersion: Version.parse(dependencyShape.version),
+            identifier: symbolName,
+            parentName: element.kind == ElementKind.FUNCTION
+                ? null
+                : parentElement.name!,
+            kind: element.kind,
+            parentKind: element.kind == ElementKind.FUNCTION
+                ? null
+                : parentElement.kind,
+            references: [span],
+          )
+        : null;
+  }
+}
+
+/// Returns the package name from a library identifier/uri, or null if
+/// [libraryUri] isn't of the form `package:*/*`.
+///
+/// Example:
+/// `packageFromLibraryUri('package:my_package/my_library.dart')` returns
+/// `'my_package'`.
+String? packageFromLibraryUri(String libraryUri) {
+  if (libraryUri.startsWith('package:') && libraryUri.contains('/')) {
+    final endIndex = libraryUri.indexOf('/');
+    return libraryUri.substring(8, endIndex);
+  }
+  return null;
+}
+
+// TODO: rename this class? it is still useful to have this structure even if we are not sure whether this is a real issue or a false positive - consider LowerBoundConstraintIssueResult or similar
+class LowerBoundConstraintIssue {
+  /// The name of the package that has an incorrect lower bound dependency constraint
+  final String dependencyPackageName;
+
+  /// The constraint on [dependencyPackageName] in the pubspec file
+  final VersionConstraint constraint;
+
+  /// The latest version of [dependencyPackageName]
+  final Version currentVersion;
+
+  /// The lowest version of [dependencyPackageName] allowed by [constraint].
+  /// This is the version which does not contain [identifier].
+  final Version lowestVersion;
+
+  /// The name of the identifier that is referenced in source code, and which is
+  /// available in the latest version of [dependencyPackageName], but not in
+  /// version [lowestVersion], which is allowed by the lower-bound dependency
+  /// constraint [constraint].
+  final String identifier;
+
+  /// The name of the enclosing class/extension of the missing [identifier], or
+  /// null if [identifier] is not a member of a class/extension.
+  final String? parentName;
+
+  // TODO: Use enumeration somehow to assert that this member is one of several supported values
+  /// The [ElementKind] of the [Element] corresponding to the missing
+  /// [identifier], one of [ElementKind.FUNCTION], [ElementKind.METHOD],
+  /// [ElementKind.GETTER], [ElementKind.SETTER].
+  final ElementKind kind;
+
+  // TODO: Use enumeration somehow to assert that this member is one of several supported values
+  /// The [ElementKind] of the parent [Element] of the missing [identifier],
+  /// [parentName], one of [ElementKind.CLASS], [ElementKind.EXTENSION], or null
+  /// if [identifier] is not a member of a class/extension.
+  final ElementKind? parentKind;
```

Comment:
Use enumeration somehow to assert that this member is one of several supported values

Reply:
create my own `ElementKind`s and enums